ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml

```diff
@@ -117,6 +117,27 @@
     lab_vars: "{{ (lab_name == 'scale') | ternary(scale, alias) }}"
   when: lab_name in ['scale', 'alias']
 
+# This works!
+- block:
+  - name: print r640s
+    set_fact:
+        machines_r640s: "{{ ocp_node_content | to_json | from_json | community.general.json_query(server_name_query) }}"
+    vars:
+      server_name_query: "nodes[?contains(pm_addr, '{{ node_string }}' )]"
+      node_string: "r640"
+
+  - name: print 5039ms
+    set_fact:
+        machines_5039ms: "{{ ocp_node_content | to_json | from_json | community.general.json_query(server_name_query) }}"
+    vars:
+      server_name_query: "nodes[?contains(pm_addr, '{{ node_string }}' )]"
+      node_string: "5039ms"
+
+  - name: set ordered ocp_node_content
+    set_fact:
+      ocp_node_content: "{ 'nodes': {{ machines_r640s | union( machines_5039ms ) }} }"
+  when: cloud_name == "cloud40"
+
```

Comment:
This is not very standard, I think we shouldn't include cloud specific blocks like this... Is there a way we can order the list of hosts dynamically depending on the bare metal server type given by an input var?, something like 

control_plane_hardware_type: r640s
workers_hardware_type: 5030ms

```suggestion
# This works!
- block:
  # If possible add an additional check that the number of control-plane nodes is 3
  - name: print control-plane-nodes
    set_fact:
        control_plane_machines: "{{ ocp_node_content | to_json | from_json | community.general.json_query(server_name_query) }}"
    vars:
      server_name_query: "nodes[?contains(pm_addr, '{{ control_plane_hardware_type }}' )]"

  - name: print worker nodes
    set_fact:
        worker_node_machines: "{{ ocp_node_content | to_json | from_json | community.general.json_query(server_name_query) }}"
    vars:
      server_name_query: "nodes[?contains(pm_addr, '{{ workers_hardware_type }}' )]"

  - name: set ordered ocp_node_content
    set_fact:
      ocp_node_content: "{ 'nodes': {{ control_plane_machines | union( worker_node_machines ) }} }"
  when: control_plane_hardware_type != ""
```

Reply:
Yeah, I hear you. My though was this is a unique configuration for cloud40, so once it worked as I proposed, I didnt' want to spend any more time on it. My thoughts on this file overall is that it already tries to be generic for too many configurations that may only be specific for a single allocation, hence this implementation.

Reply:
Incorporated and added variables with comment. Please check.